pkg/appfile/parser.go

```diff
@@ -171,9 +183,114 @@ func (p *Parser) parseTrait(name string, properties map[string]interface{}) (*Tr
 	}
 
 	return &Trait{
-		Name:     name,
-		Params:   properties,
-		Template: templ,
-		Health:   health,
+		Name:               name,
+		CapabilityCategory: capabilityCategory,
+		Params:             properties,
+		Template:           templ,
+		Health:             health,
 	}, nil
 }
+
+// GenerateApplicationConfiguration converts an appFile to applicationConfig & Components
+func (p *Parser) GenerateApplicationConfiguration(app *Appfile, ns string) (*v1alpha2.ApplicationConfiguration,
+	[]*v1alpha2.Component, error) {
+	appconfig := &v1alpha2.ApplicationConfiguration{}
+	appconfig.SetGroupVersionKind(v1alpha2.ApplicationConfigurationGroupVersionKind)
+	appconfig.Name = app.Name
+	appconfig.Namespace = ns
+	appconfig.Spec.Components = []v1alpha2.ApplicationConfigurationComponent{}
+
+	if appconfig.Labels == nil {
+		appconfig.Labels = map[string]string{}
+	}
+	appconfig.Labels[OAMApplicationLabel] = app.Name
+
+	var components []*v1alpha2.Component
+	for _, wl := range app.Workloads {
+
+		pCtx := process.NewContext(wl.Name)
```

Comment:
This part of code inside the for loop is too verbose.
Would you refactor the code to reduce the length and also reuse in getTerraformJSONFiles()

Reply:
Done, thanks.